nixos/tests/matrix-synapse.nix

```diff
@@ -48,22 +48,23 @@ in {
     {
       services.matrix-synapse = {
         enable = true;
-        database_type = "psycopg2";
-        tls_certificate_path = "${cert}";
-        tls_private_key_path = "${key}";
-        database_args = {
-          password = "synapse";
+        settings = {
+          database = {
+            name = "psycopg2";
+            args.password = "synapse";
+          };
+          tls_certificate_path = "${cert}";
+          tls_private_key_path = "${key}";
+          registration_shared_secret = registrationSharedSecret;
```

Comment:
While we're at it, I'd suggest to do something about this too. Either in the form of `passwordFile` options or the `replace-secret` + `environmentFile` technique.

Reply:
synapse has an `include` feature in its configuration, we've been using that to get secrets into synapse. it's probably a good idea to use that and have a file with all secrets generated from systemd credentials or otherwise with `jq`/`yq`. (our synapse has one secret containing a `"`, that has been a source of endless fun in the past with other deployment tools ...)

Reply:
There already is an `extraConfigFiles` option, that allows passing multiple config files to synapse. That should cover this.

Reply:
Great, so I would recommend using that in the `mkRemovedOptionModule` message.

Reply:
On second thoughts, though, It's probably better to have file option since these secrets are *shared* with other services such a coturn. coturn for instance has a `static-auth-secret-file` option for it.

Reply:
PTAL. 61e84b03409fc946b830889c58f69fb440e4b9ed

Reply:
I agree that it'd be really nice to have specifically the turn secrets be a nix-handled secret magic file script thing.

Reply:
As you found out yourself secret handling gets complicated when there is no deep-merging between multiple configuration files.

Again, I don't believe this is a regression and not every requirement should be crammed into this PR. Instead we should do incremental steps to tackle these things.